### PR TITLE
[RCTDevLoadingView] Add ability to disable during development.

### PR DESCRIPTION
**Pull request by MattFoley**
I'd like this ability as this has a tendency to get in the way of some of the more complex UI pieces I have. Disabling RCT_DEV entirely is too much for me.

### CODE REVIEW

**Comment by MattFoley** [comment]
This is a fix for this issue I opened last week. https://github.com/facebook/react-native/issues/2379

This seems to be most relevant for @nicklockwood @javache as they're the last to have worked on this class. This seemed to be the least intrusive way I could add the ability to disable/enable this at runtime.

File: React/Modules/RCTDevLoadingView.m [hunk 1]
```diff
@@ -57,59 +64,64 @@ - (void)setBridge:(RCTBridge *)bridge
...
-      source = @"pre-bundled file";
-    } else {
-      source = [NSString stringWithFormat:@"%@:%@", URL.host, URL.port];
-    }
-
-    _label.text = [NSString stringWithFormat:@"Loading from %@...", source];
-    _window.hidden = NO;
-
-  });
+  if (isEnabled) {
```
Comment (ide) [inline React/Modules/RCTDevLoadingView.m]: Just bail out & return early if `!isEnabled` to avoid all this indentation.

File: React/Modules/RCTDevLoadingView.h [hunk 2]
```diff
@@ -11,4 +11,10 @@
 
 @interface RCTDevLoadingView : NSObject <RCTBridgeModule>
 
+#if RCT_DEV
+
++ (void)setEnabled:(BOOL)enable;
```
Comment (javache) [inline React/Modules/RCTDevLoadingView.h]: Just make this `@property (nonatomic, assign, getter=isEnabled) BOOL enabled;`

File: React/Modules/RCTDevLoadingView.m (hunk 1 above)
Comment (MattFoley) [inline React/Modules/RCTDevLoadingView.m]: I prefer not to have multiple exits of a given method like that, as you end up eventually having hard to follow logic. I will make this change shortly since you've asked for it, but do you have a style guide I can reference for things like this?

Comment (ide) [inline React/Modules/RCTDevLoadingView.m]: There's no style guide but avoiding multiple exits sometimes leads to extra convoluted code. When you return early it can make your code simpler because you're reducing the state space of your program and keeping the code visually easier to read because there's less indentation.

Comment (MattFoley) [inline React/Modules/RCTDevLoadingView.m]: I'd probably break this method into somewhat smaller chunks myself. :)

**Comment by MattFoley** [comment]
Squashed and updated for feedback.
